src/commands/deploy.ts

```diff
@@ -49,7 +52,29 @@ export async function deploy(telemetryProperties: { [key: string]: string; }, tr
 
     await verifyRuntimeIsCompatible(runtime, outputChannel, client, siteWrapper);
 
-    await siteWrapper.deploy(folderPath, client, outputChannel, extensionPrefix);
+    node.treeItem.state = localize('deploying', 'Deploying...');
+    try {
+        node.refresh();
+        await siteWrapper.deploy(folderPath, client, outputChannel, extensionPrefix);
+    } finally {
+        node.treeItem.state = await node.treeItem.siteWrapper.getState(client);
+        node.refresh();
+    }
+
+    const children: IAzureNode[] = await node.getCachedChildren();
+    const functionsNode: IAzureParentNode<FunctionsTreeItem> = <IAzureParentNode<FunctionsTreeItem>>children.find((n: IAzureNode) => n.treeItem instanceof FunctionsTreeItem);
+    const functions: IAzureNode<FunctionTreeItem>[] = <IAzureNode<FunctionTreeItem>[]>await functionsNode.getCachedChildren();
```

Comment:
functionsNode is guaranteed to be non-null?

Reply:
Yep